### ISSUE TITLE
Bug: [restrict-plus-operands] `allowNumberAndString: false` doesn't report when adding a `bigint` to a `string`

### ISSUE THREAD

**Issue by 0x31** [typescript-eslint/typescript-eslint#10715]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.tsx&code=PTAEHUEMCcDsGdSQEYHsCuAXUBLWA3SAGxwBNRUAHAU2kllPgCgjrsAzWAOXQFtlaoALygAFJABcoWHwHQANKGRT4maHgDmASmEA%2BJKADUSgNxMQoACKpq8WAHJsAdxiwWbUJwBCODXmwi4lLIvv6KyqCq6rDaegbGyCZAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6aFfKsu4%2BWZLQD2paAEMmAE1QYA2uGw5E0aEOiQANAsVZFeyGPjwhAdwBysALYAjZQEEpAZS7MA5ujAAzQ8kTbsAL7%2BALoKQQFAA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
// Warns about invalid operands
let fnNumber = (a: number, b: string) => a + b;
// Doesn't warn
let fnBigint = (a: bigint, b: string) => a + b;
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/restrict-plus-operands": [
      "error",
      {
          "allowNumberAndString": false
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected that the 4th line should report the error "Operands of '+' operations must be a number or string, allowing a string + any of: `any`, `boolean`, `null`, `RegExp`, `undefined`. Got `bigint` + `string`."

The docs for `allowNumberAndString` specifically mention supporting `bigint`:
> Whether to allow bigint/number typed values and string typed values to be added together.

### Actual Result

There was no error on the 4th line.

### Additional Info

In `restrict-plus-operands.ts`, line `221` should compare `otherType` to `ts.TypeFlags.NumberLike | ts.TypeFlags.BigIntLike` instead of just `ts.TypeFlags.NumberLike`.

**Comment by ronami** [comment]
This seems like a bug to me, as the description of `allowNumberAndString` specifically mentions the `bigint` type:

> Whether to allow `bigint`/`number` typed values and `string` typed values to be added together.

I'm not 100% sure about this, so I'm waiting for another team member to mark this as "accepting prs" (unless I'm wrong on this 🙈).
